### PR TITLE
[FEAT] AoE locking

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

This PR implements the lock for AOE affected plots.

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by brian-man** [comment]
Typo in PR title

File: src/features/game/events/landExpansion/moveCrop.ts [hunk 1]
```diff
@@ -19,22 +29,74 @@ type Options = {
...
 };
 
+export function isAOELocked(
+  plot: CropPlot,
+  collectibles: Collectibles,
+  createdAt: number
+): boolean {
+  const cropName = plot.crop?.name;
+  const crop = plot.crop;
+  const isBasicCrop =
```
Comment (brian-man) [inline src/features/game/events/landExpansion/moveCrop.ts:39]: Duplicate of `isBasicCrop` in plant.ts

File: src/features/game/events/landExpansion/removeCollectible.ts [hunk 2]
```diff
@@ -82,6 +82,13 @@ export function removeCollectible({ state, action }: Options) {
     }
   }
 
+  if (action.name === "Basic Scarecrow") {
+    const collectible: PlacedLamp = collectibleToRemove;
```
Comment (brian-man) [inline src/features/game/events/landExpansion/removeCollectible.ts:86]: This logic seems related to the genie lamp.  Why is it in a scarecrow block?

File: src/features/island/collectibles/MovableComponent.tsx [hunk 3]
```diff
@@ -111,6 +111,19 @@ const canRemoveItem = (
     if (areUnsupportedChickensBrewing(state)) return false;
   }
 
+  if (name === "Basic Scarecrow") {
+    // if any crop plot is planted with Sunflower or Potato or Pumpkin and within AOE of scarecrow, return false
+    const cropPlots = Object.values(state.crops)
+      .flat()
+      .some(
+        (crop) =>
+          crop.crop?.name === "Sunflower" ||
```
Comment (brian-man) [inline src/features/island/collectibles/MovableComponent.tsx:120]: Shouldn't this use `isBasicCrop`?

File: src/features/island/collectibles/MovableComponent.tsx [hunk 4]
```diff
@@ -111,6 +111,19 @@ const canRemoveItem = (
...
+          crop.crop?.name === "Potato" ||
+          crop.crop?.name === "Pumpkin"
+      );
+
+    return !cropPlots;
+  }
```
Comment (c-py) [inline src/features/island/collectibles/MovableComponent.tsx:126]: **issue (discussed):** I think this needs to call isAOE or similar, as the scarecrow cannot be dug up unless there are no crops planted.

**Review by c-py — CHANGES_REQUESTED** [review]

**Review by c-py — APPROVED** [review]
LGTM!!

**Pull request merged**